src/controllers/tripRequestsSearchController.js

```diff
@@ -0,0 +1,224 @@
+      const finds = await tripRequest.searchByDepartureDestination(
+        departure,
+        destination,
+        req.user.id,
+        role
+      );
+      if (finds) {
+        return res.status(200).json({ status: 200, data: { trips: finds } });
+      }
+      return notFound.isNotFound(
+        { data: { message: 'Not found, try again!' } },
+        res
+      );
+    } catch (error) {
+      return applicationError.internalServerError(
+        { data: { message: `Error: ${error} ,try again!` } },
+        res
+      );
+    }
+  }
+  if (req.query.startDate && req.query.endDate) {
+    try {
+      const { startDate, endDate } = req.query;
+      const role = await rolesHelper.getUserRole(req.user.id);
+      const finds = await tripRequest.searchByStartEndDate(
+        startDate,
+        endDate,
+        req.user.id,
+        role
+      );
+      if (finds) {
+        return res.status(200).json({ status: 200, data: { trips: finds } });
+      }
+      return notFound.isNotFound(
+        { data: { message: 'Not found, try again!' } },
+        res
+      );
+    } catch (error) {
+      return applicationError.internalServerError(
+        { data: { message: `Error: ${error} ,try again!` } },
+        res
+      );
+    }
+  }
+  if (req.query.requestId) {
+    try {
+      const userRole = await rolesHelper.getUserRole(req.user.id);
+      const finds = await tripRequest.searchById(
+        req.query.requestId,
+        userRole,
+        req.user.id
+      );
+      if (finds) {
+        return res.status(200).json({ status: 200, data: { trips: finds } });
+      }
+      return notFound.isNotFound(
+        { data: { message: 'Not found, try again!' } },
+        res
+      );
+    } catch (error) {
+      applicationError.internalServerError(
+        { data: { message: `Error: ${error} ,try again!` } },
+        res
+      );
+    }
+  }
+
+  if (req.query.user) {
+    try {
+      const userRole = await rolesHelper.getUserRole(req.user.id);
+      const finds = await tripRequest.searchByUserId(
+        req.query.user,
+        userRole,
+        req.user.id
+      );
+      if (finds) {
+        return res.status(200).json({ status: 200, data: { trips: finds } });
+      }
+      return notFound.isNotFound(
+        { data: { message: 'Not found, try again!' } },
+        res
+      );
+    } catch (error) {
+      return applicationError.internalServerError(
+        { data: { message: `Error: ${error} ,try again!` } },
+        res
+      );
+    }
+  }
+
+  if (req.query.destination) {
+    try {
+      const { destination } = req.query;
+      const toLower = destination.toLowerCase();
+      const userRole = await rolesHelper.getUserRole(req.user.id);
+      const finds = await tripRequest.searchByDestination(
+        toLower,
+        userRole,
+        req.user.id
+      );
+      if (finds) {
+        return res.status(200).json({ status: 200, data: { trips: finds } });
+      }
```

Comment:
search on doing a full-text search with Postgres. This will help you reduce a lot of conditioning.